backend/api/Controllers/InspectionController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Globalization;
+using Api.Controllers.Models;
+using Api.Database.Models;
+using Api.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers
+{
+    [ApiController]
+    [Route("inspection")]
+    public class InspectionController(
+            ILogger<InspectionController> logger,
+            IInspectionService inspectionService
+        ) : ControllerBase
+    {
+
+        /// <summary>
+        /// Get the inspection image against an isarTaskId
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpGet]
+        [Authorize(Roles = Role.User)]
+        [Route("{installationCode}/{taskId}/taskId")]
+        [ProducesResponseType(typeof(Inspection), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<Inspection>> GetInspectionImageById([FromRoute] string installationCode, string taskId)
+        {
+            Inspection? inspection;
+            try
+            {
+                inspection = await inspectionService.ReadByIsarTaskId(taskId, readOnly: true);
```

Comment:
If taskId is isarTaskId, maybe it should be represented in the name to make it more readable